server/src/main/java/org/opensearch/rest/action/admin/indices/RestViewAction.java

```diff
@@ -0,0 +1,191 @@
+        }
+
+        @Override
+        public String getName() {
+            return CreateViewAction.NAME;
+        }
+
+        @Override
+        protected RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+            try (final XContentParser parser = request.contentParser()) {
+                final CreateViewAction.Request createViewAction = CreateViewAction.Request.fromXContent(parser);
+                return channel -> client.admin().indices().createView(createViewAction, new RestToXContentListener<>(channel));
+            }
+        }
+    }
+
+    public static class SearchViewHandler extends BaseRestHandler {
+        @Override
+        public List<Route> routes() {
+            return List.of(
+                new NamedRoute.Builder().path("/views/" + VIEW_ID_PARAMETER + "/_search")
+                    .method(GET)
+                    .uniqueName("cluster:views:search")
+                    .build(),
+                new NamedRoute.Builder().path("/views/" + VIEW_ID_PARAMETER + "/_search")
+                    .method(POST)
+                    .uniqueName("cluster:views:search")
+                    .build()
+            );
+        }
+
+        @Override
+        public String getName() {
+            return "view_search_action";
+        }
+
+        @Override
+        public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+            final String viewId = request.param(VIEW_ID);
+
+            final SearchViewAction.Request viewSearchRequest = new SearchViewAction.Request(viewId);
+            final IntConsumer setSize = size -> viewSearchRequest.source().size(size);
+
+            request.withContentOrSourceParamParserOrNull(
+                parser -> RestSearchAction.parseSearchRequest(
+                    viewSearchRequest,
+                    request,
+                    parser,
+                    client.getNamedWriteableRegistry(),
+                    setSize
+                )
+            );
+
+            final ValidationException validationResult = viewSearchRequest.validate();
+            if (validationResult != null) {
+                throw validationResult;
+            }
+
+            return channel -> {
+                final RestCancellableNodeClient cancelClient = new RestCancellableNodeClient(client, request.getHttpChannel());
+                cancelClient.execute(SearchAction.INSTANCE, viewSearchRequest, new RestStatusToXContentListener<>(channel));
+            };
+        }
+    }
+
+    // TODO: Replace and reorganize this layout
+
+    // public List<Route> routes() {
+
+    // return List.of(
+    // new NamedRoute.Builder().path("/views").method(GET).uniqueName("cluster:views:list").build(),
+    // new NamedRoute.Builder().path("/views/" + viewIdParameter).method(GET).uniqueName("cluster:views:get").build(),
```

Comment:
All of the APIs under 'cluster:admin/view/*` can interact with everything in the view.  Whereas in the `views:data/read/*` namespace the contents are accessible.  This will need to be interated on a little to find what minimal information is needed for the uses cases that are supported over time.

For example if a view requires parameters or a property inside of a users auth token, we should have a way to determine what the parameter is, otherwise the only way to see it is to make a request and get a failure.